spec/spec.md

```diff
@@ -0,0 +1,114 @@
+
+For the sake of simplicity, the specification only supports a single module per image.
+
+#### Running OCI Images with Envoy:
+
+Envoy supports loading and running WASM modules via a file on local disk or an “Http datasource”.
+
+Envoy WASM Filters can be stored according to the spec and run with Istio and Gloo, with the help of a local cache which pulls images from remote registries.  
+
+Control planes then configure the Envoy instances to load the filter via the local cache, using the required root_id parameter supplied in the image config if it is available.
+
+
+### Format:
+
+The WASM OCI Artifact Spec consists of two layers bundled together:
+- A layer specifying configuration for the target runtime
+- A layer containing the compiled WASM module itself
+
+Each layer is associated with its own Media Type, which is stored in the OCI Descriptor for that layer:
+
+| Media Type | Type | Description |
+|------------|------|-------------|
+| application/vnd.module.wasm.config.v1+json | JSON Object | Configuration for the Target WASM runtime.
+| application/vnd.module.wasm.content.layer.v1+wasm | binary data (byte array) | The compiled module data |
+
+`application/vnd.module.wasm.config.v1+json` Property Descriptions:
+
+| Property   | Type | Description |
+|------------|------|-------------|
+| type | string | Name of the target runtime. Required. Specifies the intended runtime of the bundled module. The content of the Opaque JSON Config is specific to the type of WASM runtime. 
+| abiVersions | string array | List of ABI Versions for the target runtime with which the image is compatible. The format for the version is dependent upon the runtime itself.
+| config | JSON Object | This field stores any configuration parameters required by the target runtime. Its structure depends on the specified runtime. |
+
+
+### Envoy WASM Filter Specification
+
+The runtime config for Envoy WASM Filter OCI Images has the following format:
+
+- *type* is set to `envoy_proxy`
+- *abiVersion* is set to a recognized version of the Envoy Proxy WASM Filter ABI 
+- *config* is a JSON Object containing a list of Filter root_ids that can be used with the provided filter
+
+The `root_ids` key in the *config* JSON Object will have a list of strings as a value. Each string in the list corresponds to the name of a registered Root Context Helper defined in the module.
+
+#### Example:
+
+The following descriptors provide an example of the OCI Image descriptors for an Envoy WASM Filter stored according to the specification:
+```
+[
+  {
+    "mediaType": "application/vnd.module.wasm.config.v1+json",
+    "digest": "sha256:d0a165298ae270c5644be8e9938036a3a7a5191f6be03286c40874d761c18abf",
+    "size": 125,
+    "annotations": {
+      "org.opencontainers.image.title": "runtime-config.json"
+    }
+  },
+  {
+    "mediaType": "application/vnd.module.wasm.content.layer.v1+wasm",
```

Comment:
This is still an odd choice for a mediaType, see https://github.com/solo-io/wasm/issues/256

Reply:
Yes this is just moved from the original one as-is. See the description and comments in the PR.